### PR TITLE
🌐 [i18n-KO] Translated `knowledge_distillation_for_image_classification.md to Korean" 

### PR DESCRIPTION
**Pull request by JinukHong**
# What does this PR do?

Translated the `knowledge_distillation_for_image_classification.md` file of the documentation to Korean.
Thank you in advance for your review. 

Part of https://github.com/huggingface/transformers/issues/20179

## Before reviewing
- [x] Check for missing / redundant translations (번역 누락/중복 검사)
- [x] Grammar Check (맞춤법 검사)
- [x] Review or Add new terms to glossary (용어 확인 및 추가)
- [x] Check Inline TOC (e.g. `[[lowercased-header]]`)
- [x] Check live-preview for gotchas (live-preview로 정상작동 확인)

## Who can review? (Initial)

@mjk0618, @cjfghk5697, @ahnjj, @JinukHong, @nuatmochoi 

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review? (Final)

@stevhliu May you please review this PR?

### CODE REVIEW

File: docs/source/ko/tasks/knowledge_distillation_for_image_classification.md [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+
+이제 데이터셋을 로드하겠습니다.
+
+```python
+from datasets import load_dataset
+
+dataset = load_dataset("beans")
+```
+
+이 경우 두 모델의 이미지 프로세서가 동일한 해상도로 동일한 출력을 반환하기 때문에, 두가지를 모두 사용할 수 있습니다. 우리는 `dataset`의 `map()` 메서드를 사용하여 데이터셋의 모든 분할마다 전처리를 적용할 것입니다.
```
Comment (cjfghk5697) [inline docs/source/ko/tasks/knowledge_distillation_for_image_classification.md:41]:
```suggestion
이 경우 두 모델의 이미지 프로세서가 동일한 해상도로 동일한 출력을 반환하기 때문에, 두가지를 모두 사용할 수 있습니다. `dataset`의 `map()` 메소드를 사용하여 데이터셋의 모든 분할마다 전처리를 적용할 것입니다.
```

Comment (cjfghk5697) [reply docs/source/ko/tasks/knowledge_distillation_for_image_classification.md:41]: glossary에 맞췄습니다!

File: docs/source/ko/tasks/knowledge_distillation_for_image_classification.md [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+
+⚠️ Note that this file is in Markdown but contain specific syntax for our doc-builder (similar to MDX) that may not be
+rendered properly in your Markdown viewer.
+
+-->
+# 컴퓨터 비전을 위한 지식 증류[[Knowledge-Distillation-for-Computer-Vision]]
+
+[[open-in-colab]]
+
+지식 증류(Knowledge distillation)는 더 크고 복잡한 모델(교사)에서 더 작고 간단한 모델(학생)로 지식을 전달하는 기술입니다. 지식을 한 모델에서 다른 모델로 증류하기 위해서는, 특정 작업(이 경우 이미지 분류)에 대해 학습된 사전 훈련된 교사 모델을 사용하고, 이미지 분류 작업을 학습할 학생 모델을 무작위로 초기화합니다. 다음으로, 학생 모델이 교사 모델의 출력을 모방하도록 하기 위해 학생 모델의 출력과 교사 모델의 출력 간의 차이를 최소화하도록 훈련합니다. 이 방법은 Hinton 등이 발표한 논문 [Neural Network에서 지식 증류](https://arxiv.org/abs/1503.02531)에서 처음 소개되었습니다. 이 가이드에서는 특정 작업에 맞춘 지식 증류를 수행할 것입니다. 이번에는 [beans dataset](https://huggingface.co/datasets/beans)을 사용할 것입니다.
```
Comment (cjfghk5697) [inline docs/source/ko/tasks/knowledge_distillation_for_image_classification.md:20]:
```suggestion
지식 증류(Knowledge distillation)는 더 크고 복잡한 모델(교사)에서 더 작고 간단한 모델(학생)로 지식을 전달하는 기술입니다. 한 모델에서 다른 모델로 지식을 증류하기 위해, 특정 작업(이 경우 이미지 분류)에 대해 학습된 사전 훈련된 교사 모델을 사용하고, 랜덤으로 초기화된 학생 모델을 이미지 분류 작업에 대해 학습합니다. 그다음, 학생 모델이 교사 모델의 출력을 모방하여 두 모델의 출력 차이를 최소화하도록 훈련합니다. 이 기법은 Hinton 등 연구진의 [Distilling the Knowledge in a Neural Network](https://arxiv.org/abs/1503.02531)에서 처음 소개되었습니다. 이 가이드에서는 특정 작업에 맞춘 지식 증류를 수행할 것입니다. 이번에는 [beans dataset](https://huggingface.co/datasets/beans)을 사용할 것입니다.
```

File: docs/source/ko/tasks/knowledge_distillation_for_image_classification.md [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+이 가이드는 [미세 조정된 ViT 모델](https://huggingface.co/merve/vit-mobilenet-beans-224) (교사 모델)을 [MobileNet](https://huggingface.co/google/mobilenet_v2_1.4_224) (학생 모델)으로 증류하는 방법을 🤗 Transformers의 [Trainer API](https://huggingface.co/docs/transformers/en/main_classes/trainer#trainer) 를 사용하여 보여줍니다.
+
+증류와 과정 평가를 위해 필요한 라이브러리를 설치해 봅시다.
+
+
+```bash
+pip install transformers datasets accelerate tensorboard evaluate --upgrade
+```
+
+이 예제에서는 `merve/beans-vit-224` 모델을 교사 모델로 사용하고 있습니다. 이 모델은 `google/vit-base-patch16-224-in21k`를 기반으로 하여 beans 데이터셋에 대해 파인 튜닝된 이미지 분류 모델입니다. 우리는 이 모델을 무작위로 초기화된 MobileNetV2로 증류해볼 것입니다.
```
Comment (cjfghk5697) [inline docs/source/ko/tasks/knowledge_distillation_for_image_classification.md:31]:
```suggestion
이 예제에서는 `merve/beans-vit-224` 모델을 교사 모델로 사용하고 있습니다. 이 모델은 beans 데이터셋에서 파인 튜닝된 `google/vit-base-patch16-224-in21k` 기반의 이미지 분류 모델입니다. 이 모델을 무작위로 초기화된 MobileNetV2로 증류해볼 것입니다.
```

File: docs/source/ko/tasks/knowledge_distillation_for_image_classification.md [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+teacher_processor = AutoImageProcessor.from_pretrained("merve/beans-vit-224")
+
+def process(examples):
+    processed_inputs = teacher_processor(examples["image"])
+    return processed_inputs
+
+processed_datasets = dataset.map(process, batched=True)
+```
+
+본질적으로, 우리는 학생 모델(무작위로 초기화된 MobileNet)이 교사 모델(파인 튜닝된 비전 트랜스포머)을 모방하도록 만들고자 합니다. 이를 달성하기 위해 먼저 교사와 학생 모델의 로짓 출력값을 구합니다. 그런 다음 각 출력을 매개변수 `temperature`로 나누는데, 이는 각 소프트 타겟의 중요성을 조절합니다. `lambda`라는 매개변수는 증류 손실의 중요성에 가중치를 줍니다. 이 예제에서는 `temperature=5`와 `lambda=0.5`를 사용할 것입니다. 우리는 학생과 교사 간의 발산을 계산하기 위해 Kullback-Leibler Divergence 손실을 사용할 것입니다. 두 데이터 P와 Q가 주어졌을 때, KL Divergence는 Q를 사용하여 P를 표현하는 데 얼만큼의 추가 정보가 필요한지를 설명합니다. 두 데이터가 동일하다면, KL Divergence는 0이며, Q로 P를 설명하는 데 추가 정보가 필요하지 않음을 의미합니다. 따라서 지식 증류의 맥락에서 KL Divergence는 유용합니다.
```
Comment (cjfghk5697) [inline docs/source/ko/tasks/knowledge_distillation_for_image_classification.md:55]:
```suggestion
학생 모델(무작위로 초기화된 MobileNet)이 교사 모델(파인 튜닝된 비전 트랜스포머)을 모방하도록 만들고자 합니다. 이를 위해, 먼저 교사와 학생 모델에서 로짓 출력값을 얻습니다. 그런 다음 각 출력을 `temperature` 매개변수로 나누어 각 소프트 타겟의 중요성을 조절합니다. `lambda`라는 매개변수는 증류 손실의 중요성에 가중치를 줍니다. 이 예제에서는 `temperature=5`와 `lambda=0.5`를 사용합니다. 학생과 교사 간의 발산을 계산하기 위해 Kullback-Leibler Divergence 손실을 사용할 것입니다. 두 데이터 P와 Q가 주어졌을 때, KL Divergence는 Q를 사용하여 P를 표현하는 데 얼마나 많은 추가 정보가 필요한지를 설명합니다. 두 데이터가 동일하다면, KL Divergence는 0이 되며, Q로 P를 설명하는 데 추가 정보가 필요하지 않음을 의미합니다. 따라서 지식 증류의 맥락에서 KL Divergence는 유용합니다.
```

File: docs/source/ko/tasks/knowledge_distillation_for_image_classification.md [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+이제 Hugging Face Hub에 로그인하여 `Trainer`를 통해 Hugging Face Hub에 모델을 푸시할 수 있도록 하겠습니다.
+
+
+```python
+from huggingface_hub import notebook_login
+
+notebook_login()
+```
+
+이제 `TrainingArguments`, 교사 모델과 학생 모델을 설정해봅시다.
```
Comment (cjfghk5697) [inline docs/source/ko/tasks/knowledge_distillation_for_image_classification.md:107]:
```suggestion
이제 `TrainingArguments`, 교사 모델과 학생 모델을 설정하겠습니다.
```

File: docs/source/ko/tasks/knowledge_distillation_for_image_classification.md [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+    metric_for_best_model="accuracy",
+    report_to="tensorboard",
+    push_to_hub=True,
+    hub_strategy="every_save",
+    hub_model_id=repo_name,
+    )
+
+num_labels = len(processed_datasets["train"].features["labels"].names)
+
+# initialize models
```
Comment (cjfghk5697) [inline docs/source/ko/tasks/knowledge_distillation_for_image_classification.md:131]:
```suggestion
# 모델 초기화
```

File: docs/source/ko/tasks/knowledge_distillation_for_image_classification.md [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+teacher_model = AutoModelForImageClassification.from_pretrained(
+    "merve/beans-vit-224",
+    num_labels=num_labels,
+    ignore_mismatched_sizes=True
+)
+
+# training MobileNetV2 from scratch
```
Comment (cjfghk5697) [inline docs/source/ko/tasks/knowledge_distillation_for_image_classification.md:138]:
```suggestion
# MobileNetV2 밑바닥부터 학습
```

File: docs/source/ko/tasks/knowledge_distillation_for_image_classification.md [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+```
+
+모델을 테스트 세트에서 평가할 수 있습니다.
+
+```python
+trainer.evaluate(processed_datasets["test"])
+```
+
+
+테스트 세트에서 우리 모델은 72%의 정확도에 도달했습니다. 증류의 효율성을 검증하기 위해 동일한 하이퍼파라미터로 beans 데이터셋에서 MobileNet을 처음부터 훈련했을 때 테스트 세트에서 63%의 정확도를 관찰했습니다. 독자들이 다양한 사전 훈련된 교사 모델, 학생 구조, 증류 매개변수를 시도하고 그 결과를 보고하도록 권장합니다. 증류된 모델의 훈련 로그와 체크포인트는 [이 저장소](https://huggingface.co/merve/vit-mobilenet-beans-224)에서 찾을 수 있으며, 처음부터 훈련된 MobileNetV2는 이 [저장소](https://huggingface.co/merve/resnet-mobilenet-beans-5)에서 찾을 수 있습니다.
```
Comment (cjfghk5697) [inline docs/source/ko/tasks/knowledge_distillation_for_image_classification.md:192]:
```suggestion
테스트 세트에서 모델은 72%의 정확도에 도달했습니다. 증류의 효율성을 검증하기 위해 동일한 하이퍼파라미터로 beans 데이터셋에 대해 처음부터 MobileNet을 훈련한 결과 테스트 세트에서 63%의 정확도를 도달했습니다. 다양한 사전 훈련된 교사 모델, 학생 구조, 증류 매개변수를 시도하고 그 결과를 공유해 주시기 바랍니다. 증류된 모델의 훈련 로그와 체크포인트는 [이 리포지토리](https://huggingface.co/merve/vit-mobilenet-beans-224)에서 찾을 수 있으며, 처음부터 훈련된 MobileNetV2는 [이 리포지토](https://huggingface.co/merve/resnet-mobilenet-beans-5)에서 찾을 수 있습니다.
```

**Review by cjfghk5697 — COMMENTED** [review]
glossary와 좀더 자연스럽게 수정해봤습니다. 고생하셨습니다!

File: docs/source/ko/tasks/knowledge_distillation_for_image_classification.md (hunk 8 above)
Comment (nuatmochoi) [reply docs/source/ko/tasks/knowledge_distillation_for_image_classification.md:192]: @cjfghk5697 마지막 line에 `리포지토` typo 있습니다:)

File: docs/source/ko/tasks/knowledge_distillation_for_image_classification.md (hunk 2 above)
Comment (ahnjj) [reply docs/source/ko/tasks/knowledge_distillation_for_image_classification.md:20]:
```suggestion
지식 증류(Knowledge distillation)는 더 크고 복잡한 모델(교사)에서 더 작고 간단한 모델(학생)로 지식을 전달하는 기술입니다. 지식을 한 모델에서 다른 모델로 증류하기 위해서는, 특정 작업(이 경우 이미지 분류)에 대해 사전 훈련된 교사 모델을 사용하고, 이미지 분류 작업을 학습할 학생 모델을 무작위로 초기화합니다. 다음으로, 학생 모델이 교사 모델의 출력을 모방하도록 하기 위해 학생 모델의 출력과 교사 모델의 출력 간의 차이를 최소화하도록 훈련합니다. 이 방법은 Hinton 등이 발표한 논문 [Neural Network에서 지식 증류](https://arxiv.org/abs/1503.02531)에서 처음 소개되었습니다. 이 가이드에서는 특정 작업에 맞춘 지식 증류를 수행할 것입니다. 이번에는 [beans dataset](https://huggingface.co/datasets/beans)을 사용할 것입니다.
```

File: docs/source/ko/tasks/knowledge_distillation_for_image_classification.md [hunk 9]
```diff
@@ -0,0 +1,192 @@
...
+
+증류와 과정 평가를 위해 필요한 라이브러리를 설치해 봅시다.
+
+
+```bash
+pip install transformers datasets accelerate tensorboard evaluate --upgrade
+```
+
+이 예제에서는 `merve/beans-vit-224` 모델을 교사 모델로 사용하고 있습니다. 이 모델은 `google/vit-base-patch16-224-in21k`를 기반으로 하여 beans 데이터셋에 대해 파인 튜닝된 이미지 분류 모델입니다. 우리는 이 모델을 무작위로 초기화된 MobileNetV2로 증류해볼 것입니다.
+
```
Comment (ahnjj) [inline docs/source/ko/tasks/knowledge_distillation_for_image_classification.md:32]:
```suggestion
이 예제에서는 `merve/beans-vit-224` 모델을 교사 모델로 사용하고 있습니다. 이 모델은 `google/vit-base-patch16-224-in21k`를 기반으로 하여 beans 데이터셋에 대해 파인 튜닝된 이미지 분류 모델입니다. 우리는 이 모델을 무작위로 초기화된 MobileNetV2에 증류해볼 것입니다.

```

**Review by ahnjj — CHANGES_REQUESTED** [review]

File: docs/source/ko/tasks/knowledge_distillation_for_image_classification.md (hunk 4 above)
Comment (ahnjj) [reply docs/source/ko/tasks/knowledge_distillation_for_image_classification.md:55]:
```suggestion
학생 모델(무작위로 초기화된 MobileNet)이 교사 모델(파인 튜닝된 비전 트랜스포머)을 모방하도록 할 것 입니다. 이를 위해 먼저 교사와 학생 모델의 로짓 출력값을 구합니다. 그런 다음 각 출력값을 매개변수 `temperature` 값으로 나누는데, 이 매개변수는 각 소프트 타겟의 중요도를 조절하는 역할을 합니다. 매개변수 `lambda` 는 증류 손실의 중요도에 가중치를 줍니다. 이 예제에서는 `temperature=5`와 `lambda=0.5`를 사용할 것입니다. 학생과 교사 간의 발산을 계산하기 위해 Kullback-Leibler Divergence 손실을 사용합니다. 두 데이터 P와 Q가 주어졌을 때, KL Divergence는 Q를 사용하여 P를 표현하는 데 얼만큼의 추가 정보가 필요한지를 말해줍니다. 두 데이터가 동일하다면, KL Divergence는 0이며, Q로 P를 설명하는 데 추가 정보가 필요하지 않음을 의미합니다. 따라서 지식 증류의 맥락에서 KL Divergence는 유용합니다.
```

File: docs/source/ko/tasks/knowledge_distillation_for_image_classification.md (hunk 8 above)
Comment (ahnjj) [reply docs/source/ko/tasks/knowledge_distillation_for_image_classification.md:192]:
```suggestion
테스트 세트에서 모델의 정확도는 72%에 도달했습니다. 증류의 효율성을 검증하기 위해 동일한 하이퍼파라미터로 beans 데이터셋에서 MobileNet을 처음부터 훈련하였고, 테스트 세트에서의 정확도는 63% 였습니다. 다양한 사전 훈련된 교사 모델, 학생 구조, 증류 매개변수를 시도해보시고 결과를 보고하기를 권장합니다. 증류된 모델의 훈련 로그와 체크포인트는 [이 저장소](https://huggingface.co/merve/vit-mobilenet-beans-224)에서 찾을 수 있으며, 처음부터 훈련된 MobileNetV2는 이 [저장소](https://huggingface.co/merve/resnet-mobilenet-beans-5)에서 찾을 수 있습니다.
```

**Review by ahnjj — CHANGES_REQUESTED** [review]

File: docs/source/ko/tasks/knowledge_distillation_for_image_classification.md [hunk 10]
```diff
@@ -0,0 +1,192 @@
...
+        self.temperature = temperature
+        self.lambda_param = lambda_param
+
+    def compute_loss(self, student, inputs, return_outputs=False):
+        student_output = self.student(**inputs)
+
+        with torch.no_grad():
+          teacher_output = self.teacher(**inputs)
+
+        # Compute soft targets for teacher and student
```
Comment (stevhliu) [inline docs/source/ko/tasks/knowledge_distillation_for_image_classification.md:83]: Just a few untranslated code comments here :)

**Review by stevhliu — APPROVED** [review]
Very nice, thanks for the contribution!

File: docs/source/ko/tasks/knowledge_distillation_for_image_classification.md (hunk 10 above)
Comment (JinukHong) [inline docs/source/ko/tasks/knowledge_distillation_for_image_classification.md:83]:
```suggestion
        #  교사와 학생의 소프트 타겟(soft targets) 계산

```

File: docs/source/ko/tasks/knowledge_distillation_for_image_classification.md [hunk 11]
```diff
@@ -0,0 +1,192 @@
...
+        soft_teacher = F.softmax(teacher_output.logits / self.temperature, dim=-1)
+        soft_student = F.log_softmax(student_output.logits / self.temperature, dim=-1)
+
+        # Compute the loss
```
Comment (JinukHong) [inline docs/source/ko/tasks/knowledge_distillation_for_image_classification.md:87]:
```suggestion
        # 손실(loss) 계산
```

File: docs/source/ko/tasks/knowledge_distillation_for_image_classification.md [hunk 12]
```diff
@@ -0,0 +1,192 @@
...
+        distillation_loss = self.loss_function(soft_student, soft_teacher) * (self.temperature ** 2)
+
+        # Compute the true label loss
```
Comment (JinukHong) [inline docs/source/ko/tasks/knowledge_distillation_for_image_classification.md:90]:
```suggestion
        # 실제 레이블 손실 계산
```

File: docs/source/ko/tasks/knowledge_distillation_for_image_classification.md [hunk 13]
```diff
@@ -0,0 +1,192 @@
...
+        student_target_loss = student_output.loss
+
+        # Calculate final loss
```
Comment (JinukHong) [inline docs/source/ko/tasks/knowledge_distillation_for_image_classification.md:93]:
```suggestion
        # 최종 손실 계산
```

File: docs/source/ko/tasks/knowledge_distillation_for_image_classification.md (hunk 10 above)
Comment (JinukHong) [reply docs/source/ko/tasks/knowledge_distillation_for_image_classification.md]: Thank you so much !!! Resolved immediately!!

**Pull request merged**